### PR TITLE
refactor: Create an interface for LSP14

### PR DESCRIPTION
**Pull request by b00ste**
## What does this PR introduce?

In this PR the ILSP14Ownable2Step is created, to be used in other contracts.

### CODE REVIEW

File: contracts/LSP14Ownable2Step/ILSP14Ownable2Step.sol [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    /**
+     * @dev emitted whenever the `renounceOwnership(..)` 2-step process is started
+     */
+    event RenounceOwnershipInitiated();
+
+    /**
+     * @dev emitted when ownership of the contract has been renounced
+     */
+    event OwnershipRenounced();
+
```
Comment (YamenMerhi) [inline contracts/LSP14Ownable2Step/ILSP14Ownable2Step.sol:23]: I suggest we add owner function in here

File: contracts/Helpers/ERC165Interfaces.sol [hunk 2]
```diff
@@ -123,10 +123,10 @@ contract CalculateLSPInterfaces {
         // prettier-ignore
         bytes4 interfaceId =
             OwnableUnset.owner.selector ^
-            OwnableUnset.transferOwnership.selector ^
-            OwnableUnset.renounceOwnership.selector ^
-            LSP14Ownable2Step.pendingOwner.selector ^
-            LSP14Ownable2Step.acceptOwnership.selector;
+            ILSP14Ownable2Step.pendingOwner.selector ^
```
Comment (YamenMerhi) [inline contracts/Helpers/ERC165Interfaces.sol:126]: You can do `OwnableUnset.owner.selector` ^ `type(ILSP14).interfaceId`

File: contracts/LSP14Ownable2Step/ILSP14Ownable2Step.sol [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+     */
+    function transferOwnership(address newOwner) external;
+
+    /**
+     * @dev Initiate the process of accepting ownership of the contract
+     */
+    function acceptOwnership() external;
+
+    /**
+     * @dev Initiate the process of renouncing ownerhsip or confirm the renouncement of the contract
```
Comment (YamenMerhi) [inline contracts/LSP14Ownable2Step/ILSP14Ownable2Step.sol:40]: My suggestion: 
> Initiate the process of renouncing ownerhsip on the first call and confirm the renouncement of the ownership on the second call

File: contracts/LSP14Ownable2Step/ILSP14Ownable2Step.sol [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+     */
+    event OwnershipRenounced();
+
+    /**
+     * @dev Returns the address of the current pending owner
+     */
+    function pendingOwner() external view returns (address);
+
+    /**
+     * @dev Initiate the process of transferring ownership of the contract
```
Comment (YamenMerhi) [inline contracts/LSP14Ownable2Step/ILSP14Ownable2Step.sol:30]: My suggestion:
> Initiate the process of transferring ownership of the contract by setting the pending owner

File: contracts/LSP14Ownable2Step/ILSP14Ownable2Step.sol [hunk 5]
```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+/**
+ * @dev Modified version of OwnableUnset contract in which, tarnsferring or renouncing ownership
```
Comment (YamenMerhi) [inline contracts/LSP14Ownable2Step/ILSP14Ownable2Step.sol:5]: transferring*

Comment (CJ42) [inline contracts/LSP14Ownable2Step/ILSP14Ownable2Step.sol:5]:
```suggestion
 * @dev Extended version of Ownable (EIP173) in which, transferring or renouncing ownership
```

File: contracts/LSP14Ownable2Step/ILSP14Ownable2Step.sol [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+ * of the contract works in 2 steps.
+ */
+interface ILSP14Ownable2Step {
+    /**
+     * @dev emitted whenever the `transferOwnership(..)` 2-step process is started
+     */
+    event OwnershipTransferStarted(address indexed previousOwner, address indexed newOwner);
```
Comment (CJ42) [inline contracts/LSP14Ownable2Step/ILSP14Ownable2Step.sol:12]: Should we have the `OwnershipTransferred` event defined in the interface as well?

@YamenMerhi @skimaharvey @b00ste

File: contracts/LSP14Ownable2Step/ILSP14Ownable2Step.sol (hunk 1 above)
Comment (CJ42) [reply contracts/LSP14Ownable2Step/ILSP14Ownable2Step.sol:23]: Agree it should be added as well.

File: contracts/LSP14Ownable2Step/ILSP14Ownable2Step.sol [hunk 7]
```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+/**
+ * @dev Modified version of OwnableUnset contract in which, tarnsferring or renouncing ownership
+ * of the contract works in 2 steps.
+ */
+interface ILSP14Ownable2Step {
+    /**
+     * @dev emitted whenever the `transferOwnership(..)` 2-step process is started
```
Comment (CJ42) [inline contracts/LSP14Ownable2Step/ILSP14Ownable2Step.sol:10]: Suggested changes.

```suggestion
     * @dev emitted when starting the `transferOwnership(..)` 2-step process.
```

File: contracts/LSP14Ownable2Step/ILSP14Ownable2Step.sol [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+     */
+    event OwnershipTransferStarted(address indexed previousOwner, address indexed newOwner);
+
+    /**
+     * @dev emitted whenever the `renounceOwnership(..)` 2-step process is started
```
Comment (CJ42) [inline contracts/LSP14Ownable2Step/ILSP14Ownable2Step.sol:15]: Suggested changes

```suggestion
     * @dev emitted when starting the `renounceOwnership(..)` 2-step process.
```

File: contracts/LSP14Ownable2Step/ILSP14Ownable2Step.sol [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+     */
+    function pendingOwner() external view returns (address);
+
+    /**
+     * @dev Initiate the process of transferring ownership of the contract
+     */
+    function transferOwnership(address newOwner) external;
+
+    /**
+     * @dev Initiate the process of accepting ownership of the contract
```
Comment (CJ42) [inline contracts/LSP14Ownable2Step/ILSP14Ownable2Step.sol:35]: I don't agree with this definition.
This does not initiate the process. It finalizes it.

```suggestion
     * @dev Complete the process of transferring ownership. MUST be called by the pendingOwner.
```

**Review by CJ42 — CHANGES_REQUESTED** [review]
added review comments

File: contracts/LSP14Ownable2Step/ILSP14Ownable2Step.sol (hunk 6 above)
Comment (skimaharvey) [reply contracts/LSP14Ownable2Step/ILSP14Ownable2Step.sol:12]: I believe it is usually a good practice to include struct and events in the interfaces

**Review by skimaharvey — APPROVED** [review]

**Review by CJ42 — APPROVED** [review]

**Pull request merged**